client/tests/integration/connected_peers.rs

```diff
@@ -46,12 +46,12 @@ fn connected_peers_with_f(faults: u64, port: u16) {
     let unregister_peer = UnregisterBox::new(IdBox::PeerId(peer.id.clone()));
     genesis_client.submit(unregister_peer).unwrap();
     thread::sleep(pipeline_time * 2);
+    thread::sleep(std::time::Duration::from_secs(10)); // New P2P system takes a little bit of time to update connections.
```

Comment:
They aren't the only ones. I'm surprised this is an improvement over the old design. And I'm not saying this just because I wrote the old `p2p` crate.